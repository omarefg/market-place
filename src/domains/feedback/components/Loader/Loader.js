import React from "react";

export function Loader() {
  return <p data-testid="Loader_title">Loading...</p>;
}
