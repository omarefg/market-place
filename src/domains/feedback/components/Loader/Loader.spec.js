import React from "react";
import { render } from "@testing-library/react";
import { Loader } from "./Loader";

describe("@domains/feedback/components/Loader", () => {
  it("Given a valid call it must render without crashing", () => {
    // Arrange
    const { container } = render(<Loader />);
    // Assert
    expect(container).toMatchSnapshot();
  });

  it("Given a valid call it must render a valid loading message", () => {
    // Arrange
    const { getByTestId } = render(<Loader />);
    const message = getByTestId("Loader_title");
    // Assert
    expect(message.innerHTML).toBe("Loading...");
  });
});
