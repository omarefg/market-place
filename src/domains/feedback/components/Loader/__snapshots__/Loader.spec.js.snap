// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@domains/feedback/components/Loader Given a valid call it must render without crashing 1`] = `
<div>
  <p
    data-testid="Loader_title"
  >
    Loading...
  </p>
</div>
`;
